docs/homepage.md

```diff
@@ -76,4 +76,71 @@ dump($automapper->map(new InputUser('John', 'Doe', 28), DatabaseUser::class));
 //   +lastName: "Doe"
 //   +age: 28
 // }
-```
\ No newline at end of file
+```
+
+### How to customize the mapping? 🚀
+
+The mapping process could be extended in multiple ways.
+
+#### Map manually a single property
+
+You can override the mapping of a single property by leveraging `AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformerInterface`.
+It can be useful if you need to map several properties from the source to a unique property in the target. 
+
+```php
+class BirthDateUserTransformer implements CustomPropertyTransformerInterface
+{
+    public function supports(string $source, string $target, string $propertyName): bool
+    {
+        return $source === InputUser::class && $target === DatabaseUser::class && $propertyName === 'birthDate';
+    }
+
+    /**
+     * @param InputUser $source
+     */
+    public function transform(object $source): \DateTimeImmutable
+    {
+        return new \DateTimeImmutable("{$source->birthYear}-{$source->birthMonth}-{$source->birthDay}");
+    }
+}
+```
+
+#### Map manually a whole object
+
+In order to customize the mapping of a whole object, you can leverage `AutoMapper\Transformer\CustomTransformer\CustomModelTransformerInterface`.
+You have then full control over the transformation between two types:
+
+```php
+use Symfony\Component\PropertyInfo\Type;
+
+class InputUserToDatabaseUserCustomTransformer implements CustomModelTransformerInterface
+{
+    public function supports(array $sourceTypes, array $targetTypes): bool
+    {
+        return $this->hasType($sourceTypes, DatabaseUser::class) && $this->hasType($targetTypes, OutputUser::class);
+    }
```

Comment:
I must maintain that I really dislike this and I think it is a terrible DX. 

From a user perspective, If I want to create a custom mapper between `DatabaseUser` and `OutputUser`, I expect the method `support()` to be passed two string, and not two arrays of strings :shrug: